text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,236 @@
+    │    ├── v1.2.3.toml
+    │    └── v1.2.3
+    └── paketo.node
+         ├── v10.20.0.toml
+         └── v10.20.0
+
+```
+
+
+### How are dependencies grouped as layers
+Layers containing dependencies are attached as the final layers to a buildpackage. The exact distribution of dependencies into layers is left up to the packaging utility.
+
+
+### How are dependency layers matched with a buildpack?
+A buildpack packaged with dependencies will have the added `depLayers` field in the `org.buildpack.buildpack.layers` metadata.
+
+``` json 
+"paketo-buildpacks/node-engine":{
+      "0.0.178":{
+         "api":"0.2",
+         "stacks":[
+            {
+               "id":"org.cloudfoundry.stacks.cflinuxfs3"
+            },
+            {
+               "id":"io.buildpacks.stacks.bionic"
+            }
+         ],
+         "layerDiffID":"sha256:9ae53c280ea60b30b8c03978fbd2f761769fe0c15e421a9ece68e2f52926ce77",
+         "depLayers":{
+              "sha256:sha256-to-dep-layer": {
+                 "path": "/cnb/deps/path/to/dep1",
+                 "id": "org.some-identifier",
+                 "version": "v1.2.3", "metadata": {}
+              },
+              "sha256:sha256-to-other-dep-layer": {
+                 "path": "/cnb/deps/path/to/dep1",
+                 "id": "paketo.node",
+                 "version": "v10.20.0",
+                 "metadata": {"cool": "beans"}
+              }
+         }
+    
+      }
+```
+Each entry in the `depLayers` object maps a layer SHA to it's path. As well as preserves the `id`, `version` and `metadata` fields from `package.toml`
+
+### How do we determine if a dependency layer should be added to a build?
+The file tree rooted at the `/cnb/deps/` is only available at `build` time. So we add all layers to a build that match the following criteria:
+  - If a buildpack passes detection all of its `depLayers` should be added to a build.
+
+### What are the contents of the `packaged.toml` file that `CNB_PATH` points to?
+
+This file looks very similar to the `json` label and can be parsed by a buildpack to look up available offline dependencies
+
+``` toml
+[[dependencies]]
+path = "/cnb/deps/some-identifier/v1.2.3"
+id = "some-identifier"
+version = "v1.2.3"
+  [dependencies.metadata]
+  # dependency-specific data
+
+[[dependencies]]
+path = "/cnb/deps/paketo.node/v1.2.3"
+id = "paketo.node"
+ID = "paketo.node"
+  [dependencies.metadata]
+  cool = "beans"
+```
+
+This file gives us a single place to read all offline dependency information during a build.
+
+
+### Other benefits?
+Build times could be further optimized, with dependencies attached as layers, there is the option to symlink a file at `cnb/dependencies/org.dep-id/stack.v1.2.3` directly to a buildpack provided layer.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- What is the strategy for staying under the 127 Layer limit.
```

Comment:
If it's an issue, packaging up the deps could be a single layer, but then you do lose some of the benefits of dedupliciation beteween layers.